cmd/triggerrun/cmd/root.go

```diff
@@ -131,55 +135,95 @@ func readTrigger(path string) ([]*v1alpha1.Trigger, error) {
 	return list, nil
 }
 
-func readHTTP(path string) (*http.Request, error) {
+func readHTTP(path string) (*http.Request, []byte, error) {
 	f, err := os.Open(path)
-
 	if err != nil {
-		return nil, fmt.Errorf("error opening file: %w", err)
+		return nil, nil, fmt.Errorf("error opening file: %w", err)
 	}
 	defer f.Close()
 
-	return http.ReadRequest(bufio.NewReader(f))
+	request, err := http.ReadRequest(bufio.NewReader(f))
+	if err != nil {
+		return nil, nil, fmt.Errorf("error reading HTTP file: %w", err)
+	}
+
+	body, err := ioutil.ReadAll(request.Body)
+	defer request.Body.Close()
+	// bodyCopy, err := request.GetBody()
+	if err != nil {
+		return nil, nil, fmt.Errorf("error reading HTTP body: %w", err)
+	}
+
+	request.Body = ioutil.NopCloser(bytes.NewReader(body))
+
+	return request, body, err
 }
 
 func getKubeClient(kubeconfig string) (triggersclientset.Interface, error) {
 	// use the current context in kubeconfig
 	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
 	if err != nil {
-		return nil, fmt.Errorf("fail to build config from the flags: %w", err)
+		return nil, fmt.Errorf("Failed to build config from the flags: %w", err)
 	}
 
 	return triggersclientset.NewForConfig(config)
 }
 
-func processTriggerSpec(client triggersclientset.Interface, tri *triggersv1.Trigger, request *http.Request, event []byte, eventID string, eventLog *zap.SugaredLogger) ([]json.RawMessage, error) {
+func processTriggerSpec(client triggersclientset.Interface, tri *triggersv1.Trigger, request *http.Request, body []byte, eventID string, eventLog *zap.SugaredLogger) ([]json.RawMessage, error) {
 	if tri == nil {
-		return nil, errors.New("Trigger is not defined")
+		return nil, errors.New("Trigger is not defined.")
 	}
 
+	//convert trigger to eventListener
 	el, err := triggersv1.ToEventListenerTrigger(tri.Spec)
 	if err != nil {
 		return nil, fmt.Errorf("fail to convert Trigger to EvenetListener: %w", err)
 	}
 
 	log := eventLog.With(zap.String(triggersv1.TriggerLabelKey, el.Name))
 
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
```

Comment:
We should move the config and client generation out of this function, and we should pass in the kubeclient as a parameter to this function. Right now, the unit test is trying to create this kubeclient using kubeconfig which is not set and failing:

```
=== RUN   Test_processTriggerSpec/testing-name
W0806 05:12:05.313777   13984 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
W0806 05:12:05.313978   13984 client_config.go:546] error creating inClusterConfig, falling back to default config: open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
{"level":"fatal","ts":1596690725.314546,"caller":"cmd/root.go:187","msg":"Failed to get in cluster config: invalid configuration: no configuration has been provided","/trigger":"","stacktrace":"github.com/tektoncd/triggers/cmd/triggerrun/cmd.processTriggerSpec\n\t/home/prow/go/src/github.com/tektoncd/triggers/cmd/triggerrun/cmd/root.go:187\ngithub.com/tektoncd/triggers/cmd/triggerrun/cmd.Test_processTriggerSpec.func1\n\t/home/prow/go/src/github.com/tektoncd/triggers/cmd/triggerrun/cmd/root_test.go:231\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:992"}
```

Reply:
In a similar theme, we should pass in the http client as well.